galley/pkg/config/analysis/analyzers/injection/injection.go

```diff
@@ -51,16 +52,28 @@ func (a *Analyzer) Metadata() analysis.Metadata {
 		Inputs: collection.Names{
 			collections.K8SCoreV1Namespaces.Name(),
 			collections.K8SCoreV1Pods.Name(),
+			collections.K8SCoreV1Configmaps.Name(),
 		},
 	}
 }
 
 // Analyze implements Analyzer
 func (a *Analyzer) Analyze(c analysis.Context) {
+	enableNamespacesByDefault := false
 	injectedNamespaces := make(map[string]bool)
 
+	// verify the enableNamespacesByDefault flag in injection configmaps
+	c.ForEach(collections.K8SCoreV1Configmaps.Name(), func(r *resource.Instance) bool {
+		if r.Metadata.FullName.Name.String() == util.InjectionConfigMap {
+			cm := r.Message.(*v1.ConfigMap)
+			enableNamespacesByDefault = GetEnableNamespacesByDefaultFromInjectedConfigMap(cm)
```

Comment:
need to consider revision info when getting the cm name.

Reply:
done, move this section to namespace loop to retrieve specified revision.